std/algorithm/comparison.d

```diff
@@ -1565,3 +1568,129 @@ unittest
                 2, "two",
                 ));
 }
+
+/**
+    Returns true if all of the elements in $(D r1) are present in $(D r2).
+    Otherwise, returns false.
+
+    If the passed ranges are random access ranges, then this function will
+    not allocate GC handled memory. But, if your parameters are forward ranges,
+    then this function can allocate if the $(D AllocateGC.yes) enum is passed.
+    This has the benefit of being much faster than the $(D AllocateGC.no) option.
+
+    Params:
+        r1 = Can be any finite forward or random access range
+        r2 = Can be any finite forward or random access range
+
+    Returns:
+        $(D bool)
+*/
+enum AllocateGC
```

Comment:
Also, I wonder if this enum should be placed in a more general module, since it's pertinent not just to `std.algorithm.sorting`, but to any Phobos function that might want to give the user a choice of whether or not to allocate. In any case, it also needs its own ddoc header.

Alternatively, we could consider using `Flag!"allocateGC"` (from std.typecons) instead of inventing a new enum for what's simply an on/off switch.